lib/change_stream.js

```diff
@@ -476,96 +475,73 @@ function waitForTopologyConnected(topology, options, callback) {
   }, 3000); // this is an arbitrary wait time to allow SDAM to transition
 }
 
-// Handle new change events. This method brings together the routes from the callback, event emitter, and promise ways of using ChangeStream.
-function processNewChange(args) {
-  const changeStream = args.changeStream;
-  const error = args.error;
-  const change = args.change;
-  const callback = args.callback;
-  const eventEmitter = args.eventEmitter || false;
-  const cursor = changeStream.cursor;
-
-  // If the cursor is null, then it should not process a change.
-  if (cursor == null) {
-    // We do not error in the eventEmitter case.
-    if (eventEmitter) {
-      return;
-    }
+// Wait for topology to re-connect, then create a new cursor.
+// If in eventEmitter mode, returns null and doesn't iterate the change stream.
+function resumeFromError({ changeStream, callback, eventEmitter }) {
```

Comment:
is this code change necessary for the goals of converting to use `maybePromise`? It's a little hard to read the changes here

Reply:
also, FWIW I mildly prefer not using this type of "passing a whole named object in as a single parameter" thing that was refactored into this code about a year ago. Do you think this style has any explicit benefits?

Reply:
No, this refactor wasn't necessary, I just thought it was cleaner to move the retry into a function, so e.g. it shows up explicitly in call stacks.

The only reason I like the destructuring style is it makes parameter names explicit when you're calling a function with a bunch of optional arguments, and reduces code that checks the number and types of arguments passed in to figure out which is which - but, that's not the case here. I just did it this way to stay consistent with the parent function. Would you prefer I inline the `resumeFromError` code back into `processNewChange`, or switch to a normal argument style?

Reply:
I don't know why but the pattern has always rubbed me a bit wrong. It introduces a needless temporary object allocation (on top of `arguments`) and can lead to the same types of ambiguity (we still aren't checking if the passed in object exists here, for instance). I don't have very strong feelings about this, but those are my 2 cents.